### ISSUE TITLE
CLI: Implement `klareco inspect tensor` command for decoding embeddings

### ISSUE THREAD

**Issue by marctjones** [marctjones/klareco#639]
## Description

Implement command to decode tensor embeddings to human-understandable information (similar words).

**Critical for development**: When debugging models, we need to understand what embeddings represent.

Part of **Epic #637** - CLI Phase 1: Inspection Tools

## Usage

```bash
# Decode tensor annotation
klareco inspect tensor ast.json --key root_embedding --node verbo

# Show top-10 similar roots
klareco inspect tensor ast.json --key root_embedding --node verbo --top-k 10

# Visualize embedding space (2D projection)
klareco inspect tensor ast.json --key root_embedding --visualize

# Input from pipe
klareco annotate text "La hundo vidis la katon." | klareco inspect tensor --key root_embedding --node verbo
```

## Output Format

```
=== Tensor Decoding: root_embedding (verbo) ===
Root: vid
Embedding: torch.tensor([0.12, -0.34, 0.56, ...]) (64d)

Most similar roots:
  1. rigar (0.89) - to look at, to gaze
  2. observ (0.84) - to observe
  3. spekti (0.81) - to watch
  4. vidi (0.79) - to see (variant)
  5. rigard (0.77) - look, gaze (noun)
  ...

Can't fully decode: This is a learned semantic representation.
Use --similar to explore semantic neighborhood.
```

## Implementation

Location: `klareco/cli/inspect.py`

1. Read AST from file or stdin (JSON)
2. Navigate to specified node path
3. Extract tensor annotation
4. If root embedding: Look up similar roots using RootEmbeddingsAnnotator.get_similar_roots()
5. If other tensor: Show shape, statistics, warn that it can't be fully decoded
6. Support visualization with matplotlib/plotly (optional dependency)

## Key Functions

```python
def decode_root_embedding(tensor, model_path, top_k=10):
    """Decode root embedding to similar words."""
    annotator = RootEmbeddingsAnnotator(model_path=model_path)
    # Find root for this embedding
    # Get similar roots
    # Format output

def decode_tensor_generic(tensor):
    """Generic tensor info (shape, stats, can't fully decode)."""
    return {
        'shape': tensor.shape,
        'dtype': tensor.dtype,
        'mean': tensor.mean(),
        'std': tensor.std(),
        'message': 'This is a learned representation that cannot be fully decoded to text.'
    }
```

## Testing

- Test root embedding decoding
- Test generic tensor decoding
- Test node path navigation
- Test error handling (missing node, missing annotation, not a tensor)

## Dependencies

- RootEmbeddingsAnnotator (klareco.embeddings.root_annotator)
- Trained root embeddings model
- klareco.ast_annotator utility functions

Blocked by: #638 (need inspect ast first)
Part of: Epic #637

**Comment by marctjones** [comment]
Closing as part of a 2026-09-08 backlog cleanup (see PLAN.md, DESIGN.md, CLAUDE.md's merge gate).

This issue predates the merge gate (adopted 2026-07-13) and the AST-native orchestrator rewrite. It describes work against the pre-orchestrator RAG pipeline or the learned model stack (M1/M2 embeddings, compositional embeddings, entity classifiers, neural rerankers) that was pruned from the repo -- klareco/embeddings/, klareco/models/, and klareco/summarization/ no longer exist. It has no milestone and was never benchmarked, matching exactly the failure mode CLAUDE.md documents (25 capability issues filed 2026-05-26, 8 implemented, none benchmarked).

Labeled archived-2026-09 so this batch is recoverable: gh issue list --state closed --label archived-2026-09. If this capability is still wanted, re-file it against the current architecture naming the benchmark metric and test the merge gate requires.
